build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""
+upload_logs_to_s3.py
+
+Uploads build logs and index.html to an S3 bucket using the AWS CLI via subprocess.
+Intended for use in CI (e.g., GitHub Actions) at the end of a build, regardless of success/failure.
+"""
+
+import os
+import glob
+import subprocess
+import sys
+
+
+def log(*args, **kwargs):
+    # Wrapper for print that flushes immediately.
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def run_aws_cp(src, dest, content_type=None):
+    """
+    Upload a file or directory to S3 using the AWS CLI (`aws s3 cp`).
+    Args:
+        src: Source file or directory path.
+        dest: Destination S3 URI.
+        content_type: Optional MIME type to set in S3 metadata.
+    """
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
```

Comment:
nit: could check that `aws` is available as a command to run using https://docs.python.org/3/library/shutil.html#shutil.which and provide a good error message if missing, or use the AWS Python SDK instead of the CLI.

Reply:
Sounds good added a check using `shutil`